src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/service/AssessmentService.kt

```diff
@@ -325,4 +331,59 @@ class AssessmentService(
       policy = policy,
     )
   }
+
+  @Transactional
+  fun getContacts(prisonNumber: String): AssessmentContactsResponse {
+    val currentAssessment = this.getCurrentAssessment(prisonNumber)
+
+    // Currently we do not support Prison offender manager
+    val contactTypes = listOf(UserRole.PRISON_CA, UserRole.PRISON_DM, UserRole.PROBATION_COM)
+    val agents = getAgents(currentAssessment, contactTypes)
+
+    val contacts = agents.map {
+      with(it) {
+        ContactResponse(fullName, role, getEmailAddress(it), getLocationName(it))
+      }
+    }
+
+    return AssessmentContactsResponse(contacts)
+  }
+
+  private fun getAgents(
+    currentAssessment: Assessment,
+    contactTypes: List<UserRole>,
+  ): List<Agent> {
+    val contacts = currentAssessment.assessmentEvents.sortedByDescending { it.eventTime }.filter { it.agent.role in contactTypes }
+      .distinctBy { it.agent.role }.map { it.agent }
+    return contacts
+  }
+
+  private fun getEmailAddress(agent: Agent): String? {
+    try {
+      with(agent) {
+        return when (role) {
+          UserRole.PROBATION_COM -> probationService.getStaffDetailsByUsername(username)?.email
+          UserRole.PRISON_CA, UserRole.PRISON_DM -> managedUsersService.getEmail(username)?.email
+          else -> null
+        }
+      }
+    } catch (e: ItemNotFoundException) {
```

Comment:
Aren't you coercing 404s to null rather than exceptions?

Reply:
Yes because I don't want it to all fail to show other information because we cant get an email, I want the record to still display what we have? I log it!